hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -721,8 +721,7 @@ int QuectelNcpClient::configureApn(const CellularNetworkConfig& conf) {
         netConf_ = networkConfigForImsi(buf, strlen(buf));
     }
     // FIXME: for now IPv4 context only
-    auto resp = parser_.sendCommand("AT+CGDCONT=1,\"IP\",\"%s%s\"",
-            (netConf_.hasUser() && netConf_.hasPassword()) ? "CHAP:" : "",
+    auto resp = parser_.sendCommand("AT+CGDCONT=1,\"IP\",\"%s\"",
             netConf_.hasApn() ? netConf_.apn() : "");
     const int r = CHECK_PARSER(resp.readResult());
```

Comment:
@avtolstoy While this is a good solution, why not disable hasUser and hasPassword? Because hasUser and hasPassword are being returned in cellular_credentials_get(), and if we are not going to use only user and password, we might convey that information through the API call also?

Reply:
I did not get any of that :sweat_smile: What do you mean by 'disable hasUser and hasPassword'?

Reply:
In this line - 
> (netConf_.hasUser() && netConf_.hasPassword()) ? "CHAP:" : "",

So if user and password are true for the module, we use a "CHAP:" in the APN name. So if we are completely not going to user name and password for quectel modules or b5soms, then instead of removing that line, we can just set hasUser and hasPasswrod to false, no?

Additionally for example, we are also using hasUser here in cellular_credentials_get(). If we are not going to use username and password for quectel modems on b5soms altogether, we can set them false, and that way the users can access that info from the API call also.
https://github.com/particle-iot/device-os/blob/77f129979ad51544690da04976ad6992c5261cfa/hal/network/ncp/cellular/cellular_hal.cpp#L239

Reply:
We can get rid of them here for quectel / b5soms?? 

https://github.com/particle-iot/device-os/blob/77f129979ad51544690da04976ad6992c5261cfa/hal/network/ncp/cellular/network_config_db.cpp#L39

Reply:
like add a #define or whichever is the best way.

Reply:
Keerthy and I discussed why this is not needed in a short meeting. Marking as resolved.